Added status progress and miscellaneous fixes

Gustash opened the pull request:
This PR does a few things:
- Added empty stubs to the iOS JS implementation to match the Android one, to avoid having to check the platform each time.
- Added installUpdate function to start the native install implementation for Flexible updates.
- Status listener returning bytes downloaded and total bytes to download.
- Fixed listeners not working due to a mismatched `this` context.

Here's a video with a rough implementation of these changes:

![Example Flow](https://user-images.githubusercontent.com/8539174/88419625-6db0ef00-cddd-11ea-814e-389db852368b.gif)

If you want me to write documentation for these let me know.

Also let me know if you'd be interested in making me a maintainer of this package as well.

Gustash commented:
I would advise releasing this as at least a minor version update, because the listener changes are breaking changes.

SudoPlz commented:
This looks good, can you please add documentation as well?
And I'll merge and release in a new minor version.

Also, I'll be happy to join efforts in maintaining this package.

Thank you.

Gustash commented:
@SudoPlz I'll update this PR with documentation soon. 

Thanks for adding me as a maintainer as well!

Gustash commented:
@SudoPlz just added documentation.

I also added some description for each method, and added documentation for the progress listeners (didn't add for the other listener as I'm not sure what it refers to).

Also added `maintainers` to the `package.json`, hope you don't mind 😊

SudoPlz commented:
Looks great, of course i don't mind man.

Gustash commented:
I don't think I can publish this to NPM, so feel free. I don't want to intrude too much on your project anyway, so I'll leave releases to you

SudoPlz commented:
No worries, released under 0.1.0, thanks for the PR I appreciate it.
